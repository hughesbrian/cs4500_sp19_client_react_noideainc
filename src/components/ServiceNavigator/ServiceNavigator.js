import React from 'react'
import ServiceCategoryList from './ServiceCategoryList'
import ServiceCategorySectionList from './ServiceCategorySectionList'
import SearchBarContainer from '../SearchBar/SearchBarContainer';

const ServiceNavigator = ({serviceCategories, history}) =>
    <div>
        <div className="row">
            <div className="col-8">
                <SearchBarContainer history={history}/>
            </div>
            <div className="col-3 text-right">
                <a href="#">Sign up</a>
            </div>
            <div className="col-1">
                <a href="#">Log in</a>
            </div>
        </div>
        <br/>
        <br/>
        <div className="row">
            <div className="col-3">
                <ServiceCategoryList
                    serviceCategories={serviceCategories}/>
            </div>
            <div className="col-9">
                <ServiceCategorySectionList
                    serviceCategories={serviceCategories}/>
            </div>
        </div>
    </div>

export default ServiceNavigator
