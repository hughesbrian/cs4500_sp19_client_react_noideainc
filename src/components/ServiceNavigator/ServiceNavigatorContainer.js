import React from 'react'
import _ from 'lodash'
import ServiceNavigator from './ServiceNavigator'
import ServiceCategoryService from '../../services/ServiceCategoryService'
import ServiceService from '../../services/ServiceService'

class ServiceNavigatorContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceCategoryService = ServiceCategoryService.getInstance();
        this.serviceService = ServiceService.getInstance();
        this.state = {
            serviceCategories: []
        }
    }

    componentDidMount() {
        this.findAll()
    }

    findAll = () =>
        this.serviceCategoryService
            .findAllServiceCategories()
            .then(serviceCategories =>
                this.serviceService
                    .findAllServices()
                    .then(services => {
                        serviceCategories.forEach(sc =>
                            sc.services = services.filter(s =>
                                _.filter(s.serviceCategories, ['title', sc.title]).length > 0
                            ))

                        this.setState({
                            serviceCategories: serviceCategories,
                        })
                    })
            )

    render = () =>
        <ServiceNavigator serviceCategories={this.state.serviceCategories} history={this.props.history}/>

}

export default ServiceNavigatorContainer
