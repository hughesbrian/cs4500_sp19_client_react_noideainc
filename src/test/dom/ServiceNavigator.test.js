import React from 'react'
import ServiceNavigatorContainer from '../../components/ServiceNavigator/ServiceNavigatorContainer'
import TestRenderer from 'react-test-renderer'
import sleep from '../util/sleep'
import '../mockservice/ServiceNavigator.mock'

test('ServiceNavigatorContainer renders correctly', async () => {
    const testRenderer = TestRenderer.create(
        <ServiceNavigatorContainer/>
    )
    await sleep(1000) // sleep to give the component a chance to render.

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})
