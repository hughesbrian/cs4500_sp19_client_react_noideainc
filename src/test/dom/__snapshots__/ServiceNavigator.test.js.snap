// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceNavigatorContainer renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <div
        className="input-group input-group-lg"
      >
        <input
          className="form-control"
          placeholder="Search for providers"
          type="text"
        />
        <input
          className="form-control"
          placeholder="Zip code"
          type="text"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <br />
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <ul
        className="list-group"
      >
        <li
          className="list-group-item no-border category-side-list"
        >
          <a
            href="#1"
          >
            Tutoring
          </a>
        </li>
        <li
          className="list-group-item no-border category-side-list"
        >
          <a
            href="#2"
          >
            Fitness Trainer
          </a>
        </li>
        <li
          className="list-group-item no-border category-side-list"
        >
          <a
            href="#3"
          >
            House Painter
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      >
        <li
          className="list-group-item no-border service-category-sections"
        >
          <div>
            <a
              id={1}
            />
            <h2>
              Tutoring
            </h2>
            <div>
              <div
                className="row"
              >
                <div
                  className="card no-border col-6"
                >
                  <img
                    className="card-img-top service-card"
                    src="https://picsum.photos/300/200"
                  />
                  <div
                    className="card-body"
                  >
                    <p
                      className="card-text"
                    >
                      <a
                        href="/providers"
                      >
                         
                        cleaning1
                      </a>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6 list-group-item no-border service-links"
                >
                  <a
                    href="/providers"
                  >
                     
                    cleaning1
                  </a>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className="list-group-item no-border service-category-sections"
        >
          <div>
            <a
              id={2}
            />
            <h2>
              Fitness Trainer
            </h2>
            <div>
              <div
                className="row"
              >
                <div
                  className="card no-border col-6"
                >
                  <img
                    className="card-img-top service-card"
                    src="https://picsum.photos/300/200"
                  />
                  <div
                    className="card-body"
                  >
                    <p
                      className="card-text"
                    >
                      <a
                        href="/providers"
                      >
                         
                        cleaning1
                      </a>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-6 list-group-item no-border service-links"
                >
                  <a
                    href="/providers"
                  >
                     
                    cleaning1
                  </a>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className="list-group-item no-border service-category-sections"
        >
          <div>
            <a
              id={3}
            />
            <h2>
              House Painter
            </h2>
            <div>
              <div
                className="row"
              />
              <div
                className="row"
              />
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
